### DISCUSSION TITLE
Running a Erigon full node with receipts and tx history only

### DISCUSSION THREAD

**Discussion by miohtama** [erigontech/erigon#6612] [Q&A] [answered]
Erigon provides multiple command line history options for storing history:

```
 --prune value                             Choose which ancient data delete from DB:
                                             h - prune history (ChangeSets, HistoryIndices - used by historical state access, like eth_getStorageAt, eth_getBalanceAt, debug_traceTransaction, trace_block, trace_transaction, etc.)
                                             r - prune receipts (Receipts, Logs, LogTopicIndex, LogAddressIndex - used by eth_getLogs and similar RPC methods)
                                             t - prune transaction by it's hash index
                                             c - prune call traces (used by trace_filter method)
                                             Does delete data older than 90K blocks, --prune=h is shortcut for: --prune.h.older=90_000
                                             Example: --prune=hrtc (default: "disabled")
```

However this help and documentation is not very understandable. For example, what is the parameter to store the full history (no prune).

What would the correct Erigon settings to run the node that is equal to `geth --txlookuplimit 0` i.e. store all transactions and receipts, but not the historical state. 

This would be a full node with the receipt history, but not an archive node.

**Comment by AskAlexSharov** [comment, marked as answer]
no `--prune` means no prune :-) 
erigon is archive node by default. 

if "need store receipts and txs (txn lookup index)" - it means "need prune everything else: call traces and historical state": `--prune=hc`

**Reply by miohtama** [reply]
Thank you @AskAlexSharov .

Also if I understand correctly these settings cannot be changed after the node has created its initial database?

**Reply by AskAlexSharov** [reply]
Yes
